Patrick Kennedy

mixelpixel opened the pull request:
- match header style for Stretch Tasks from last week's curriculum.
- console.logging assignment description

SunJieMing commented on src/arrays.js:
```diff
@@ -9,37 +9,87 @@ const each = (elements, cb) => {
-  // Return the new array.
-};
 
 const reduce = (elements, cb, startingValue) => {
```
You can also do this for a starting value: `(elements, cb, startingValue = elements.shift())` for the default value.  This takes advantage of the ES6 default parameters.

SunJieMing replied:
Unless of course you are avoiding modifying the input array

SunJieMing commented on src/arrays.js:
```diff
@@ -9,37 +9,87 @@ const each = (elements, cb) => {
   // Elements will be passed one by one into `cb`.
   // `startingValue` is the starting value.  If `startingValue` is undefined then make `elements[0]` the initial value.
+  if (startingValue === undefined) {
+    startingValue = (typeof elements[0]) === 'string' ? '' : 0;
```
Just make `startingValue` the first item from the array

SunJieMing replied:
A reduce is supposed to have the first item in the array as its initial memo

SunJieMing commented on src/arrays.js:
```diff
@@ -9,37 +9,87 @@ const each = (elements, cb) => {
+const map = (elements, cb) => {
+  // Produces a new array of values by mapping each value in list through a transformation function (iteratee).
+  // Return the new array.
+  return reduce(elements, (acc, v) => {
```
Really creative

invegat replied:
thanks

SunJieMing commented on src/arrays.js:
```diff
@@ -9,37 +9,87 @@ const each = (elements, cb) => {
 /* Extra Credit */
-
+function isArray(elements) {
+  return (Object.prototype.toString.call(elements) === '[object Array]');
```
The preferred way of checking whether a variable is an array or not is this: `Array.isArray(myVariable);`

SunJieMing replied:
Good job creating another function though

SunJieMing commented on src/callbacks.js:
```diff
@@ -24,58 +24,77 @@
 // code here
 
 const foods = ['pineapple', 'mango', 'ribeye', 'curry', 'tacos', 'ribeye', 'mango'];
+function firstItem(foods,f) {
```
I'd prefer it if you put a space between the parameters

invegat replied:
accustomed  to  getting a 'comma-spacing' error by the eslint configs I have been working with

SunJieMing commented on src/callbacks.js:
```diff
@@ -24,58 +24,77 @@
 
 const foods = ['pineapple', 'mango', 'ribeye', 'curry', 'tacos', 'ribeye', 'mango'];
+function firstItem(foods,f) {
+  return f(foods[0])
```
This line should have a semicolon

SunJieMing commented on src/callbacks.js:
```diff
@@ -24,58 +24,77 @@
 // code here
-
+function getLength(foods,f) {
+  return f(foods.length)
```
You're being very consistent with everything though which is really good.  Consistency is huge.  Inconsistent code is a painful mark of an inexperienced developer.  It shows experience and discipline to keep your code consistent.

SunJieMing commented on src/objects.js:
```diff
@@ -1,26 +1,47 @@
+  return Object.keys(obj);
 };
-
+function isFunction(func) {
```
For the sake of consistency this should be written the same way the other functions are written.

SunJieMing commented on tests/arrays.test.js:
```diff
@@ -66,7 +66,7 @@ describe('arrays', () => {
       const result = arrayMethods.reduce(arr, callBackMockFn);
       expect(result).toBe(25);
-      expect(callBackMockFn.mock.calls.length).toBe(4);
+      expect(callBackMockFn.mock.calls.length).toBe(5);
```
This test should not be modified.  The memo in the reduce should be the first item from the array and so the callback would only be invoked 4 times.

mixelpixel committed:
Stretch Task header

- match header style for Stretch tasks from last week's curriculum.
- console.logging assignment decription
